### DISCUSSION TITLE
🍄XSOSI 计划 1 月份数据

### DISCUSSION THREAD

**Discussion by will-ww** [X-lab2017/open-wonderland#236] [2️⃣ 聊项目]
2022 年的 XSOSI 总共运行了 7 期：
- [6 月份数据](https://github.com/orgs/X-lab2017/discussions/115)
- [7 月份数据](https://github.com/X-lab2017/open-wonderland/discussions/149)
- [8 月份数据](https://github.com/X-lab2017/open-wonderland/discussions/173)
- [9 月份数据](https://github.com/X-lab2017/open-wonderland/discussions/196)
- [10 月份数据](https://github.com/X-lab2017/open-wonderland/discussions/210)
- [11 月份数据](https://github.com/X-lab2017/open-wonderland/discussions/221)
- [12 月份数据](https://github.com/X-lab2017/open-wonderland/discussions/231)

2023 年的第 1 期，正值春节，总体上有一个下降的趋势与现实也非常相符合：

![image](https://user-images.githubusercontent.com/15010826/216207581-4ec84f0c-f7b3-47fe-a4b5-05bcc19ceb22.png)
![image](https://user-images.githubusercontent.com/15010826/216207161-bef579a0-26a3-4669-8974-58249d585fc4.png)

有一个新的关注点大家需要注意到：一月份活跃的实际上只有 12 个人，另外 12 个是因为衰减的贡献度导致的，这个也可以讨论一下，对于某个月不活跃的情况，如何来设计分配机制。例如，在阿里的实践是，当月没有贡献是不分配的，但保留原来 OpenRank 的 80%，次月再有贡献时可以继承（感谢 @frank-zsy 带来的实践与信息）。如果采用这样的方式，就会得到如下图所示的 1 月份的分配方案：

![image](https://user-images.githubusercontent.com/15010826/216206740-28c35a6f-61c7-4d18-a4cc-8b85e7d03b16.png)

个人比较倾向于这种鼓励回来连续参与贡献的导向。而在 24 人大名单的分配版本中，没有考虑到这个，当月衰减的 80% 也是按照贡献来计算了，所以这个相当于是保留一部分名誉，下次贡献时继承，不贡献的话就持续衰减了，即 1 月份的活跃人数确认为 12 人。

<img width="719" alt="8a901762a045431459d8a5f4d52cadb" src="https://user-images.githubusercontent.com/15010826/216208024-75d7f886-4670-4800-815f-0260d52f16e2.png">

**Comment by will-ww** [comment]
XSOSI 大屏已同步更新（感谢 @zhicheng-ning ）：http://dataease.nzcer.cn/link/E1OQKVBz

![image](https://user-images.githubusercontent.com/15010826/216208486-05d9ab9a-a20a-44f0-b6e2-79627ca05e81.png)

**Comment by will-ww** [comment]
同步，[OpenLeaderboard ](https://open-leaderboard.x-lab.info/)也已更新，同样能够得到的观察：中国的开发者过年的影响还是很明显的~ :-)

<img width="1346" alt="86744c585b7d17b975410330291209d" src="https://user-images.githubusercontent.com/15010826/216208890-dc0dc46f-208c-48d5-b6ba-aa928c8bd7b2.png">

**Comment by frank-zsy** [comment]
由于最近在做 OpenRank 可视化的工作，在细节拆分中发现了一些在计算中的 bug，主要问题是会使得 Issue 和 PR 作者的 OpenRank 偏高，在修改 bug 后重新计算了所有相关项目的 OpenRank 情况。

考虑到 2022.12 与 2023.1 的 XSOSI 补贴尚未发放，更新了这两个月的数据，总体如下：

<img width="1412" alt="image" src="https://user-images.githubusercontent.com/8512426/217404999-84f51988-cabf-43b3-b7d8-52841e09bcc8.png">

整体的浮动较小，变化不大，但整体提升了非 Issue 和 PR 作者的参与者的贡献比例。

目前大家可以用[这个 CodePen](https://codepen.io/frank-zsy/pen/abjMXBV) 来观察某项目某月的具体 OpenRank 的价值流转情况，可以看到每个开发者、每个协作单元的价值即转移比例等，大家也可以一起来捉虫，看是否还存在有问题的地方。目前我个人观察下来没发现其他问题，大家有什么好的建议意见也可以随时提出。

**Reply by tyn1998** [reply]
12月的已经发放了~

**Reply by will-ww** [reply]
> 12月的已经发放了~

问题应该不大，@bifenglin 后续多关注，发放的时候核对下，差别大记录好后续补回来~

**Reply by bifenglin** [reply]
没问题，新的OpenRank方式自2023年1月份开始。
